### DISCUSSION TITLE
Wrapping part of a View onto another page, but ensuring that title & body text always appear on the same page?

### DISCUSSION THREAD

**Discussion by clarencefoy** [diegomura/react-pdf#2626] [Q&A]
Hi all,

I'm loving this library, very impressed by it. I have a question for page-wrapping.

Imagine I have a View component which nests two things, a title and a body text, like so:

```
<View > // <------- can't set this component to wrap=false, as that causes lots of unused space on pages
        <Text></Text> // <----- this heading needs to be on the same page as the body text below it
        <Text></Text> // <---- this text can wrap across pages
</View>
```
![image](https://github.com/diegomura/react-pdf/assets/77112468/2ee37028-cd7e-4f34-9cf5-b2aaa02f1b11)

The body-text can be quite long, so I would quite like it to be able to wrap onto the next page. However, at times the title gets disconnected from the body text, as the page wraps exactly between the title and the body text.

Is there any way to ensure both that the title always stays connected to the body text, but the body text also wraps onto the next page? 

Disabling page wrapping for the whole text component causes huge amounts of white space in my documents, as all components now sit on their own page, so I'd like to avoid that if possible.

**Comment by thaoluon** [comment]
Hello, Nice to meet you.
Please tell me how to make to split into multi pages when content is large.
Please show me opensource sample code.
Please Please

**Comment by kyle-robinson-morpheus** [comment]
Also interested in this. [Here's an example I'm working with.](https://react-pdf.org/repl?code=3187b0760ce02e004d09e01b02984605e18194a4e5600b6592803a6002764043299002806f00a06180076a0736402e1859b369c00988809660bbf006c001958c00be006915d001e51fa0a1016da852e93f8046004c6a84c0066e0a167100bcf8c530058ad0cd50020a2712e307e0072002b005768711b7810af363b302800316a3d714478508015713d54005a002510033010c5556625004a006e66665048581110600c187a45001e00110e8882e4803e45365e80056e64385c147446041408524e1e253867647400225f773d981164002308ae71eb4980357164007779bc25c6180323491c90767e00567f9c9944a6b8dc0090bd7b93c60e21112cf62d2835124c80a1ed41131ba4c72c82d28c00f21476011a890181e3d2ec1429060390a3c060c052549245c0e0517e30001bb511011642f400f4b8fc563b15087b3c56bb4623024106a751b230108d8501a788c0ce200a29c28fc3da99d81a0c5288560ec4e2f1b01952c656b1f16d7524b244096000e4c55a6e0051300c02201fe5710c2739a21a830002c99399d41538699117808060603299ca8e9b4c011c220f4e4c1a8a03d6180a898ccc149e2ab5e822ccb4d401e7a761e6053cf137200b751a0f56e17cae1a74e3044080d0a3c8d809c51e4007a8d1440a13989ea0890ca1daf62c9cdae04413304e701bb01f5496d38cd0e21874cdd551da145a40154c0519032ea3a3a1da6f40043b4179600a37cd906dc6e60228300d110113080f9761c46000067e0d170f8009cd99541131100571cd046540acd7709cc7322203fcce75c270487d31de73a0c274208cecd16822035c6f000be034e1c70c8ab6fc0047f5ce10882828d58f6904e1d38f03ac168206197b342d0280a8341380a0e849ce6653ab12c1e300375a57d4408f540a011d904fcef301807e554b819510014a11476e56cc30e844d80c41673d2795b3f976838430a4f426c079a07425a380a270b92089d5322a7373e8fd1c4080f4583c8c9d7b20b18ecc20641ca28d4aeadc0493a8fb212e0b975d5c424dc70c22077312490405a5713017b0a40303380f830337d8b03060503c2f836af4ad06000f2a19243d1920c2a900b3491a80eb268ecab65d9050d6aa92c8af3245f26055a035ca9c1b2992a52f0e12359d1329b176ec76f9c204c9235ab6ca3c3a53cef5ca606b20c5ab230e0400ddf56a169148286cba8086e71869330005681391da0841c6f34de2d638049016c62a36a184da229492935015b0d5c45432ce270f52b405a70c0e2e89f5e7489aca4d392c6407a8602e124ebbd70a12e1a2284bbc6c08eec8780d9d69001140b345e770d89d97409dae83eae70b2200da96ccdb351dd5f11988dc6034dd5d41d728de829ab5ab2d35251921cd69da241b135873c439d83323a8089accd376b994e680e661ad354128710ce79d6a35c8329adb0a076968c212024f42bcc17bb0c956381a3e419881a0dfd4a3f2ec748c26ab798b7c45f60608a5a00cb32d855b5d591287d9a8c31ab28b52792617b545c07d76d0f4d0c49389a66440012e2caad2af826f6da7bd1a3820df9a9ca322e61f7385514a04b46e615a1478afc98855beafe14661e0c16150605a464bf980689a62868300280531fa0a45204b5f53d0768c00ff9000000)

My goal is to make that the `View` with id `container` will start on the new page if there isn't X amount of space available on the new page.

In my actual application, I'm rendering a single page and allowing react-pdf to handle the page wrapping as it sees fit. My report layout is also configurable so I can't really just add page breaks where after each new component because like the original poster said, there will be lots of whitespace between report sections.

I have tried toying with `orphans` (which works fine for text) and `minPresenceAhead` but can't seem to make a combination of the two work if I want a whole `<View>` to begin on the next page if there isn't enough space.
